### PR TITLE
Improve global env

### PR DESCRIPTION
**Pull request by jwerle**

### CODE REVIEW

File: api/internal/monkeypatch.js [hunk 1]
```diff
@@ -169,7 +170,13 @@ export function init () {
...
     FileSystemFileHandle,
     FileSystemDirectoryHandle,
-    FileSystemWritableFileStream
+    FileSystemWritableFileStream,
+
+    // buffer
+    Buffer,
+
+    // platform detection
+    isSocketRuntime: true
```
Comment (chicoxyzzy) [inline api/internal/monkeypatch.js:179]: we can also make it non-configurable, but it's not very important I guess

File: test/src/webview.js [hunk 2]
```diff
@@ -1,11 +1,111 @@
...
+test('globalThis.onload', async (t) => {
+  t.ok(await loaded, 'globalThis.onload called')
+})
+
+test('globalThis.isSocketRuntime', (t) => {
+  t.ok(globalThis.isSocketRuntime === true, 'globalThis.isSocketRuntime')
+})
+
+test('globalThis.Buffer', (t) => {
+  t.ok(typeof globalThis.Buffer === 'function' && globalThis.Buffer === Buffer, 'globalThis.Buffer')
```
Comment (chicoxyzzy) [inline test/src/webview.js:38]: I'd change the message here and in other places to something like "Buffer is available on globalThis" and "globalThis.URL is normalized"

**Review by chicoxyzzy — APPROVED** [review]

File: test/src/webview.js (hunk 2 above)
Comment (jwerle) [reply test/src/webview.js:38]: we can improve this later 🙏🏼

File: api/internal/monkeypatch.js (hunk 1 above)
Comment (jwerle) [reply api/internal/monkeypatch.js:179]: The `install()` function already does that

**Pull request merged**